From dede71919a3e322aaae641946443ffb51cc1fdf9 Mon Sep 17 00:00:00 2001
From: Jonas Witschel <diabonas@archlinux.org>
Date: Sat, 17 Oct 2020 17:18:06 +0200
Subject: [PATCH] test_x509: add tzinfo to not_before and not_after datetimes

When using asn1crypto >= 1.0.0, the unit test fails with the following error
message otherwise:

ERROR: test_self_sign_certificate (tests.test_x509.X509Tests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/build/python-pkcs11/src/python-pkcs11-0.7.0/tests/__init__.py", line 95, in wrapper
    return func(self, *args, **kwargs)
  File "/build/python-pkcs11/src/python-pkcs11-0.7.0/tests/test_x509.py", line 186, in test_self_sign_certificate
    'not_before': Time({
  File "/usr/lib/python3.8/site-packages/asn1crypto/core.py", line 1165, in __init__
    raise e
  File "/usr/lib/python3.8/site-packages/asn1crypto/core.py", line 1157, in __init__
    value = spec(value, **params)
  File "/usr/lib/python3.8/site-packages/asn1crypto/core.py", line 1677, in __init__
    raise e
  File "/usr/lib/python3.8/site-packages/asn1crypto/core.py", line 1669, in __init__
    self.set(value)
  File "/usr/lib/python3.8/site-packages/asn1crypto/core.py", line 5008, in set
    raise ValueError('Must be timezone aware')
ValueError: Must be timezone aware
    while constructing asn1crypto.core.UTCTime
    while constructing asn1crypto.x509.Time

(cherry picked from commit d693ce7cb9f245534cf5ab848d2ea830abcdbd19)
---
 requirements.in    | 2 +-
 requirements.txt   | 6 +++---
 tests/test_x509.py | 4 ++--
 3 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/requirements.in b/requirements.in
index 1ad788c..a50e080 100644
--- a/requirements.in
+++ b/requirements.in
@@ -1,3 +1,3 @@
 aenum; python_version < "3.6"
-asn1crypto
+asn1crypto>=1.0.0
 cached-property
diff --git a/requirements.txt b/requirements.txt
index 3b0908c..2e40a93 100644
--- a/requirements.txt
+++ b/requirements.txt
@@ -2,8 +2,8 @@
 # This file is autogenerated by pip-compile
 # To update, run:
 #
-#    pip-compile --output-file requirements.txt requirements.in
+#    pip-compile --output-file=requirements.txt requirements.in
 #
 aenum==2.0.7 ; python_version < "3.6"
-asn1crypto==0.22.0
-cached-property==1.3.0
+asn1crypto==1.4.0
+cached-property==1.5.2
diff --git a/tests/test_x509.py b/tests/test_x509.py
index 796ae45..0807995 100644
--- a/tests/test_x509.py
+++ b/tests/test_x509.py
@@ -184,10 +184,10 @@ class X509Tests(TestCase):
             },
             'validity': {
                 'not_before': Time({
-                    'utc_time': datetime.datetime(2017, 1, 1, 0, 0),
+                    'utc_time': datetime.datetime(2017, 1, 1, 0, 0, tzinfo=datetime.timezone.utc),
                 }),
                 'not_after':  Time({
-                    'utc_time': datetime.datetime(2038, 12, 31, 23, 59),
+                    'utc_time': datetime.datetime(2038, 12, 31, 23, 59, tzinfo=datetime.timezone.utc),
                 }),
             },
             'subject_public_key_info': {
-- 
2.38.0

